### PR TITLE
Add `BlogArticleCategory` support

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

- Add `BlogArticleCategoryListBlock` component
- Update `ArticleCategoryList` visual component
- Fetch blog article categories from Storyblok for relevant pages
- Resolve and parse categories on `blog-article` content type pages (for teaser)

## Justify why they are needed

More support for the blog.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2462** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2462" target="_blank">Graphite</a> 
    * **PR #2466** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2466" target="_blank">Graphite</a> 
      * **PR #2472** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2472" target="_blank">Graphite</a> 
        * **PR #2475** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2475" target="_blank">Graphite</a> 
          * **PR #2476** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2476" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2476?utm_source=stack-comment).

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2462** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2462" target="_blank">Graphite</a> 
    * **PR #2466** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2466" target="_blank">Graphite</a> 
      * **PR #2472** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2472" target="_blank">Graphite</a> 
        * **PR #2475** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2475" target="_blank">Graphite</a> 
          * **PR #2476** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2476" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2476?utm_source=stack-comment).

**Review by alebedev — APPROVED** [review]

File: apps/store/src/services/storyblok/storyblok.ts [hunk 1]
```diff
@@ -363,14 +365,28 @@ const BLOG_ARTICLE_CONTENT_TYPE = 'blog-article'
...
   })
 
-  return response.data.stories as Array<BlogArticleStory>
+  const items = response.data.stories as Array<RawBlogArticleStory>
+
+  return items.map<BlogArticleStory>((item) => ({
+    ...item,
+    content: {
+      ...item.content,
+      categories: item.content.categories.reduce((acc, storyId) => {
```
Comment (alebedev) [inline apps/store/src/services/storyblok/storyblok.ts:378]: Can be simplified by replacing with `.map(...).filter(Boolean)`

File: apps/store/src/services/storyblok/storyblok.ts [hunk 2]
```diff
@@ -363,14 +365,28 @@ const BLOG_ARTICLE_CONTENT_TYPE = 'blog-article'
...
+        const category = response.data.rels.find((rel) => rel.uuid === storyId)
```
Comment (alebedev) [inline apps/store/src/services/storyblok/storyblok.ts:379]: If we have many categories, should we index by category Id for fast lookup?

**Review by alebedev — APPROVED** [review]

File: apps/store/src/services/storyblok/storyblok.ts (hunk 2 above)
Comment (robinandeer) [reply apps/store/src/services/storyblok/storyblok.ts:379]: agreed, this only happens server-side at build time tho so perhaps we can postpone that optimization

File: apps/store/src/services/storyblok/storyblok.ts (hunk 1 above)
Comment (robinandeer) [reply apps/store/src/services/storyblok/storyblok.ts:378]: hmm good point, perhaps I did this to make typescript happy 🤔

**Comment by robinandeer** [comment]
@robinandeer started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2476).

**Comment by robinandeer** [comment]
[Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2476) rebased this pull request as part of a merge.

**Pull request merged**